Added more features to cups_lpadmin

HP41 opened the pull request:
Added support for classes.
Class mode or Printer mode - It must be specified and will decide how the module interprets and works with the data provided.

Added more documentation as per [official module creation guidelines](http://docs.ansible.com/ansible/developing_modules.html#conventions-recommendations), added some PEP 8 best practices.

Thanks for having a look!

HP41 committed:
Added some best practices as per PEP 8. Added a whole bunch of methods to help with handling classes in CUPS. Also add a few features for printers such as cupsIPPSupplies and cupsSNMPSupplies.

HP41 committed:
Added more documentation and lots more code comments. Also set class_members to type list and its default to [].

HP41 committed:
As class_members can now have an empty array by default, changed the checking of if it's empty to a 'not'.

HP41 committed:
Added RESULTS documentation as per module guidelines. Ensured all arguments in argument_specs have 'required' defined

HP41 committed:
Set printer to be default in printer_or_class. Reflected this in the documentation. Fixed some documentation typos. Reduced the logic when state=absent as removing printers or classes use the same commands.

HP41 committed:
Added documentation to each method, general info about the class and its inner workings. Removed prefixing a driver with '/' when no '://' is found in it as it was causing issues and not needed as the correct driver has to be provided anyways.

HP41 committed:
Changed some doc string documentation.

k0ste commented:
Thanks.

HP41 commented:
We should PR to the multimac and then to ansible-modules-extra?

k0ste commented:
@HP41, may be create another PR. Now I think we need remove code about uri. I think Administrator should define protocol, not module.

``` python
        # Use lpd if a protocol is not specified
        if self.uri and ':/' not in self.uri:
            self.uri = 'lpd://{0}/'.format(self.uri)
```

HP41 commented:
I very much agree. I believe @multimac wrote that in. Maybe he can explain as to why he did it?

k0ste commented:
@HP41, because when I found module, he have very basic feature.
See new [commits](https://github.com/k0ste/ansible-modules-extras/commits/newfeature), add purge (pre deploy) function. If u want - create PR for delete uri code.

HP41 commented:
That's a great functionality! I can help clean some stuff out and try to get code towards python best practices. 
As for `check_mode` should we leave that for the end as we need to change up a lot of code. First let's clean up and bring everything upto spec and then go for check mode?

multimac commented:
Hey, awesome work you guys have done.

There's no real reason for 'lpd' over anything else, just that at the time it was just what I was using the most. I agree the code shouldn't be there though.

k0ste commented:
@HP41, yes we wait refactoring of my bullshit code :)

HP41 commented:
Hahaha...no-one had bullshit code. We helped each other quite well! 👍

HP41 commented:
Guys, @k0ste @multimac, I pulled all the changes and updated [HP41/ansible-modules-extras](https://github.com/HP41/ansible-modules-extras/tree/newfeature). I've added you guys as members/contributors. Let's finish this once and for all and submit it to ansible? :D
